tests/test_log_encoder.py

```diff
@@ -13,22 +13,16 @@
 # limitations under the License.
 
 import unittest
-import pytest
 from typing import List, Tuple
 
-from opentelemetry.sdk._logs import SeverityNumber
-from opentelemetry.exporter.otlp.proto.http.trace_exporter.encoder import (
+from opentelemetry._logs import SeverityNumber
+from opentelemetry.exporter.otlp.proto.common._internal import (
     _encode_attributes,
     _encode_span_id,
     _encode_trace_id,
     _encode_value,
 )
-from snowflake.telemetry._internal.encoder.otlp.proto.common.log_encoder import (
-    _encode_logs,
-)
-from snowflake.telemetry._internal.exporter.otlp.proto.logs import (
-    _ProtoLogExporter,
-)
+from opentelemetry.exporter.otlp.proto.common._log_encoder import encode_logs
```

Comment:
What do we test here?

Reply:
We're using the mock data helpfully created here to:

1. Make sure that the behavior of our custom 1.12.0-based _encode_logs() method matches the behavior of the new 1.23.0 encode_logs() method from the opentelemetry lib
2. Exercise _ProtoLogExporter._serialize_logs_data() to do a basic sanity check on the bytes it produces